File: mobile/lib/models/voucher.dart
class VoucherModel {
  final String id;
  final String code;
  final String userId;
  final String? userName;
  final String? supportTicketId;
  final double amount;
  final String reason;
  final String status;
  final String? approvedByAdminId;
  final DateTime? issuedAt;
  final DateTime? expiresAt;
  final DateTime? redeemedAt;
  final DateTime createdAt;

  VoucherModel({
    required this.id,
    required this.code,
    required this.userId,
    this.userName,
    this.supportTicketId,
    required this.amount,
    required this.reason,
    required this.status,
    this.approvedByAdminId,
    this.issuedAt,
    this.expiresAt,
    this.redeemedAt,
    required this.createdAt,
  });

  factory VoucherModel.fromJson(Map<String, dynamic> json) {
    return VoucherModel(
      id: json['id'] ?? '',
      code: json['code'] ?? '',
      userId: json['userId'] ?? '',
      userName: json['userName'],
      supportTicketId: json['supportTicketId'],
      amount: (json['amount'] as num?)?.toDouble() ?? 50.0,
      reason: json['reason'] ?? 'Goodwill compensation',
      status: json['status'] ?? 'Draft',
      approvedByAdminId: json['approvedByAdminId'],
      issuedAt: json['issuedAt'] != null ? DateTime.tryParse(json['issuedAt']) : null,
      expiresAt: json['expiresAt'] != null ? DateTime.tryParse(json['expiresAt']) : null,
      redeemedAt: json['redeemedAt'] != null ? DateTime.tryParse(json['redeemedAt']) : null,
      createdAt: DateTime.tryParse(json['createdAt'] ?? '') ?? DateTime.now(),
    );
  }
}

File: backend/Models/Voucher.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Travyle.Api.Models;

public enum VoucherStatus
{
    Draft,
    Active,
    Redeemed,
    Expired,
    Revoked
}

public class Voucher
{
    [Key]
    public Guid Id { get; set; } = Guid.NewGuid();

    [Required]
    [MaxLength(50)]
    public string Code { get; set; } = string.Empty;

    [Required]
    public Guid UserId { get; set; }
    public User? User { get; set; }

    public Guid? SupportTicketId { get; set; }
    public SupportTicket? SupportTicket { get; set; }

    [Column(TypeName = "decimal(18,2)")]
    public decimal Amount { get; set; } = 50.00m;

    [MaxLength(200)]
    public string Reason { get; set; } = "Goodwill compensation";

    public VoucherStatus Status { get; set; } = VoucherStatus.Draft;

    public Guid? ApprovedByAdminId { get; set; }

    public DateTime? IssuedAt { get; set; }
    public DateTime? ExpiresAt { get; set; }
    public DateTime? RedeemedAt { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}
